### DISCUSSION TITLE
Get the list of AdHoc workflows

### DISCUSSION THREAD

**Discussion by cesar-moya** [Netflix/conductor#3017] [Ideas]
## What are you missing in the docs
How can I build a search that gives me the list of AdHoc workflows (this is, all workflow executions that do not have a workflowDef registered on the system)

## Proposed text
To add such search query in the docs for future reference

**Comment by jxu-nflx** [comment]
@cesar-moya what is your use case for this? Currently we don't support search for only adhoc workflows. One workaround you could do is to set `correlationId` for all your adhoc workflows, and use that in the search query.

**Reply by cesar-moya** [reply]
@jxu-nflx We don't want to register all testing workflows our users do while sandboxing and trying things out, therefore we ask them to create adHoc workflows to do quick tests in the system without registering the metadata workflow, so that we keep only a clean list of registered workflows at any given time. 
These users sometimes need to see which workflows they have executed, I found that I actually would need to filter based on 2 properties: whether it is an AdhocWorkflow, and also the user that executed it (which is also missing in our conductor version).
If I set the correlationId, is this property indexed through elasticsearch?

**Reply by jxu-nflx** [reply]
yes, `correlationId` is indexed: https://github.com/Netflix/conductor/blob/main/es6-persistence/src/main/resources/mappings_docType_workflow.json#L4
